modules/programs/himalaya.nix

```diff
@@ -0,0 +1,100 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.programs.himalaya;
+
+  enabledAccounts =
+    lib.filterAttrs (_: a: a.himalaya.enable) (config.accounts.email.accounts);
+
+  tomlFormat = pkgs.formats.toml { };
+
+  himalayaConfig = let
+    toHimalayaConfig = account:
+      {
+        email = account.address;
+        name = account.realName;
+        default = account.primary;
+
+        # FIXME: does not support disabling TLS altogether
+        # NOTE: does not accept sequence of strings for password commands
+        imap-login = account.userName;
+        imap-passwd-cmd = builtins.toString account.passwordCommand;
+        imap-host = account.imap.host;
+        imap-port = account.imap.port;
+        imap-starttls = account.imap.tls.useStartTls;
+
+        smtp-login = account.userName;
+        smtp-passwd-cmd = builtins.toString account.passwordCommand;
```

Comment:
```suggestion
        smtp-passwd-cmd = toString account.passwordCommand;
```

Reply:
Actually, now that I think about it `escapeShellArgs` would be better, for special characters (the command is given straight to `sh -c` by `himalaya`).